tools/semgrep.yml

```diff
@@ -0,0 +1,51 @@
+# See https://github.com/returntocorp/semgrep/blob/experimental/docs/config/advanced.md
+
+rules:
+####################### PYTHON RULES #######################
+  - id: deprecated-render-usage
+    pattern: django.shortcuts.render_to_response(...)
+    message: "Use render() (from django.shortcuts) instead of render_to_response()"
+    languages: [python]
+    severity: ERROR
+
+  - id: useless-if-body
+    patterns:
+      - pattern: |
+          if $X:
+              $S
+          else:
+              $S
+    message: "Useless if statment; both blocks have the same body"
+    languages: [python]
+    severity: ERROR
+
+  - id: dont-use-stream-objects-filter
+    pattern: Stream.objects.filter(...)
```

Comment:
Could this be instead `Stream.objects.*`?  I think that's the rule we actually want; there's a similar `Stream.objects.get` rule we should move too.  And I'd like to create a near-identical rule for `UserProfile.objects`; though that one will require moving a bit of code to `zerver/lib/users.py` first (or some temporary excludes).

It's probably worth doing a more complete

Reply:
Hi there, just wanted to check in from r2c and make sure things are going well with the move to semgrep. To achieve `Stream.objects.*` you can simply use `Stream.objects` - this will catch `Stream.objects.filter` and `Stream.objects.get`. For example:

```yaml
$ cat rule.yaml 
rules:
  - id: test-id
    pattern: Stream.objects
    message: "Test message"
    languages: [python]
    severity: ERROR
```

```
$ cat test.py 
Stream.objects.get('foo')
Stream.objects.filter('bar')
```

```
$ docker run --rm -v "${PWD}:/home/repo" returntocorp/semgrep --config rule.yaml
test.py
rule:test-id: Test message
1:Stream.objects.get('foo')
2:Stream.objects.filter('bar')
```

Reply:
Thanks @mschwager for the detailed example.

Reply:
Should this be `zerver.models.Stream.objects`?

Reply:
Yeah, I suppose this will be better.